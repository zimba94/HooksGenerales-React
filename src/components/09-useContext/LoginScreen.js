import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const { setUser } = useContext(UserContext);
    return (
        <div>
            <h1>Login Screen</h1>
            <hr></hr>
            <button
                    className="btn btn-primary"
                    onClick={()=> setUser({
                        id: 123,
                        name: "Alan"
                    })}
            >
                Login   
            </button>
        </div>
    )
}
