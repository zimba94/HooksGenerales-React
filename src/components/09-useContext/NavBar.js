import React from 'react';
import { NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (

        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <div className="container-fluid">
                <NavLink to="./" className="navbar-brand">Use Context</NavLink>
                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                    <ul className="navbar-nav me-auto mb-2 mb-lg-0">
                    <li className="nav-item">
                        <NavLink exact activeClassName="active" className="nav-link active" to="./">Home</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact activeClassName="active" className="nav-link" to="./about">About</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact activeClassName="active" className="nav-link" to="./login">Login</NavLink>
                    </li>
                    </ul>
                </div>
            </div>
        </nav>
    )
}
