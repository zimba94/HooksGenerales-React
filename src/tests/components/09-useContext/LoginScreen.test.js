import React from 'react';
import { mount} from 'enzyme';
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <LoginScreen />', () => {

   const setUser = jest.fn();
   
   const wrapper = mount(
        <UserContext.Provider value={{setUser}}>
            <LoginScreen />
        </UserContext.Provider>
   );
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe ejecutar el setUser con el argumento esperado', () => {
        wrapper.find('button').prop('onClick')();
        expect(setUser).toHaveBeenCalledWith({
            id: 123,
            name: 'Alan'
        });
    });
    
    
});
