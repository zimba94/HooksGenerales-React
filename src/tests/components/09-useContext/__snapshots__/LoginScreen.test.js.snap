// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
